views/HtmlStatic.html

```diff
@@ -1,41 +1,79 @@
-    {{#each partyItems as party}}
-      <ElectionItem item='{{party}}' hasTrendSpace='{{hasTrendSpace}}'/>
-    {{/each}}
+  <Header title='{item.title}' hideTitle='{hideTitle}' subtitle='{item.subtitle}' withErrorMargin='{withErrorMargin}' hasErrorMargin='{hasErrorMargin}' errorMarginLabels='{errorMarginLabels}'/>
+  <div class="q-election-parties {thresholdOffsetClasses}">
+    <Threshold threshold='{threshold}' base='{base}' withErrorMargin='{withErrorMargin}'/>
+    {#each partyItems as party}
+      {#if withErrorMargin}
+        <ElectionItemErrorMargin item='{party}' maxErrorMarginValue='{maxErrorMarginValue}'/>
+      {:else}
+        <ElectionItem item='{party}' hasTrendSpace='{hasTrendSpace}'/>
+      {/if}
+    {/each}
   </div>
-  {{#if otherPartiesItem}}
+  {#if otherPartiesItem}
     <div class="q-election-others">
-      <ElectionItem item='{{otherPartiesItem}}' hasTrendSpace='{{hasTrendSpace}}'/>
+      {#if withErrorMargin}
+        <ElectionItemErrorMargin item='{otherPartiesItem}' maxErrorMarginValue='{maxErrorMarginValue}'/>
+      {:else}
+        <ElectionItem item='{otherPartiesItem}' hasTrendSpace='{hasTrendSpace}'/>
+      {/if}
     </div>
-  {{/if}}
-  <Footer sources='{{item.sources}}' notes='{{item.notes}}' updatedDate='{{item.updatedDate}}' hideUpdatedDate='{{hideUpdatedDate}}'/>
+  {/if}
+  <Footer sources='{item.sources}' notes='{item.notes}' updatedDate='{item.updatedDate}' hideUpdatedDate='{hideUpdatedDate}'/>
 </div>
 
 <script>
   import Header from './Header.html';
   import Threshold from './Threshold.html';
   import ElectionItem from './ElectionItem.html';
+  import ElectionItemErrorMargin from './ElectionItemErrorMargin.html';
   import Footer from './Footer.html';
 
   export default {
     computed: {
-      parties: (item) => {
+      parties: ({ item }) => {
         return item.parties;
       },
-      hideTitle: (toolRuntimeConfig) => {
+      hideTitle: ({ toolRuntimeConfig }) => {
         return toolRuntimeConfig.displayOptions && toolRuntimeConfig.displayOptions.hideTitle;
       },
-      hasCurrentResults: (parties) => {
+      hasCurrentResults: ({ parties }) => {
         let currentResults = parties.filter(party => {
           return party.percentage !== undefined;
         });
         return currentResults.length > 0;
       },
-      sortedParties: (parties) => {
+      hasCurrenterrorMargins: ({ parties }) => {
+        let currentResults = parties.filter(party => {
+          return party.errorMargin !== undefined && party.errorMargin.bestGuess !== undefined;
+        });
+        return currentResults.length > 0;
+      },
+      sortedParties: ({ parties, item }) => {
         if (parties !== undefined) {
+          if (item.withErrorMargin) {
+            return parties.sort(function(partyA, partyB) {
+              // if any of the 2 parties has no errorMargin, we sort by this first
+              if (!partyA.errorMargin && !partyB.errorMargin) {
```

Comment:
Should be sorted based on bestGuess as well. See ->

![Bildschirmfoto 2019-03-20 um 16 27 58](https://user-images.githubusercontent.com/1810384/54697029-2ce00080-4b2d-11e9-9dac-b0a1c84b4303.png)

Reply:
this gets sorted by bestGuess only now.